File: easy/sumofleftleaves.cpp
#include <iostream>

class Solution {
public:
    int sumOfLeftLeaves(TreeNode* root) {
        if(root == NULL) {
            return 0;
        }

        int sum = 0;
        if(root->left != NULL && root->left->left == NULL && root->left->right == NULL) {
            sum += root->left->val;
        }

        sum += sumOfLeftLeaves(root->left);
        sum += sumOfLeftLeaves(root->right);
        return sum;
    }
};

File: python/easy/sumofleftleaves.py
class Solution:
    def sumOfLeftLeaves(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0

        sum1 = 0
        if root.left and not root.left.left and not root.left.right:
            sum1 += root.left.val

        sum1 += self.sumOfLeftLeaves(root.left)
        sum1 += self.sumOfLeftLeaves(root.right)

        return sum1